core/src/main/scala/akka/persistence/jdbc/state/DurableStateQueries.scala

```diff
@@ -47,43 +47,36 @@ import akka.persistence.jdbc.config.DurableStateTableConfiguration
     durableStateTable.filter(_.persistenceId === persistenceId)
 
   private[jdbc] def insertDbWithDurableState(row: DurableStateTables.DurableStateRow, seqNextValue: String) = {
-    sqlu"""INSERT INTO #${durableStateTableCfg.tableName}
-            (
-             #${durableStateTableCfg.columnNames.persistenceId},
-             #${durableStateTableCfg.columnNames.globalOffset},
-             #${durableStateTableCfg.columnNames.revision},
-             #${durableStateTableCfg.columnNames.statePayload},
-             #${durableStateTableCfg.columnNames.stateSerId},
-             #${durableStateTableCfg.columnNames.stateSerManifest},
-             #${durableStateTableCfg.columnNames.tag},
-             #${durableStateTableCfg.columnNames.stateTimestamp}
-            )
-            VALUES
-            (
-              ${row.persistenceId},
-              #${seqNextValue},
-              ${row.revision},
-              ${row.statePayload},
-              ${row.stateSerId},
-              ${row.stateSerManifest},
-              ${row.tag},
-              #${System.currentTimeMillis()}
-            )
-      """
+    durableStateTable += row.copy(globalOffset = seqNextValue, stateTimestamp = System.currentTimeMillis())
   }
 
   private[jdbc] def updateDbWithDurableState(row: DurableStateTables.DurableStateRow, seqNextValue: String) = {
-    sqlu"""UPDATE #${durableStateTableCfg.tableName}
-           SET #${durableStateTableCfg.columnNames.globalOffset} = #${seqNextValue},
-               #${durableStateTableCfg.columnNames.revision} = ${row.revision},
-               #${durableStateTableCfg.columnNames.statePayload} = ${row.statePayload},
-               #${durableStateTableCfg.columnNames.stateSerId} = ${row.stateSerId},
-               #${durableStateTableCfg.columnNames.stateSerManifest} = ${row.stateSerManifest},
-               #${durableStateTableCfg.columnNames.tag} = ${row.tag},
-               #${durableStateTableCfg.columnNames.stateTimestamp} = ${System.currentTimeMillis}
-           WHERE #${durableStateTableCfg.columnNames.persistenceId} = ${row.persistenceId}
-             AND #${durableStateTableCfg.columnNames.revision} = ${row.revision} - 1
-        """
+    durableStateTable
+      .filter(t =>
+        t.persistenceId === row.persistenceId && t.revision === (row.revision - 1)
+      )
+      .map(t =>
+        (
+          t.globalOffset,
+          t.revision,
+          t.statePayload,
+          t.stateSerId,
+          t.stateSerManifest,
+          t.tag,
+          t.stateTimestamp
+        )
+      )
+      .update(
+        (
+          seqNextValue,
+          row.revision,
+          row.statePayload,
+          row.stateSerId,
+          row.stateSerManifest,
+          row.tag,
+          System.currentTimeMillis()
+        )
+      )
```

Comment:
I'm quite sure there was a reason why we used the string based statements.

I think the schema is optional so you can probably construct it with something like 
```
schema.map(_ + ".").getOrElse("") + table
```

Reply:
I know that the current implementation won't work as the `seqNextVal` is a string, but the global offset is a long (which would need to be addressed). I will try to revert to the string statements, but would be interested as to the the reasoning for these statements since the resulting sql should be much the same.

The issue using `.map.getOrElse` is the interpolation and requirement of `#` in some cases that wouldn't allow a blank schema, but I will give it a try.

Reply:
I have restored the string sql statements and updated it such that it should maintain the `schema.table` literal created from the fold statement on the optional schema.